src/charm.py

```diff
@@ -196,22 +207,45 @@ def __init__(self, *args):
             self.on.check_config_action, self._on_check_config  # pyright: ignore
         )
 
-    def set_ports(self):
-        """Open necessary (and close no longer needed) workload ports."""
-        planned_ports = {
-            OpenedPort("tcp", self._ports.api),
-            OpenedPort("tcp", self._ports.ha),
-        }
-        actual_ports = self.unit.opened_ports()
+        self.framework.observe(self.on.collect_unit_status, self._on_collect_unit_status)
 
-        # Ports may change across an upgrade, so need to sync
-        ports_to_close = actual_ports.difference(planned_ports)
-        for p in ports_to_close:
-            self.unit.close_port(p.protocol, p.port)
+    def _on_collect_unit_status(self, event: CollectStatusEvent):
+        # Resource limit
+        if not self.resources_patch.is_ready():
+            event.add_status(
+                MaintenanceStatus("[resource patch] Waiting for resource limit patch to apply")
```

Comment:
When designing collect-status, we thought about including a prefix/component name automatically, but decided against it as status messages should be nicely-designed UI for the user. This kind of shows why: the fact that it's resource patch-related is already clear from the message part, so the `[resource patch]` prefix just adds noise to the UI. Something to consider...